src/compas/files/ply.py

```diff
@@ -138,106 +141,98 @@ def read_header(self):
         # read it as text
         # otherwise file.tell() can't be used reliably
         # to figure out where the header ends
-        with open(self.filepath) as file:
-            file.seek(0)
-
-            line = file.readline().rstrip()
-
-            if line != 'ply':
-                raise Exception('not a valid ply file')
-
-            self.start_header = file.tell()
-
-            element_type = None
+        if next(self.content).rstrip() != 'ply':
```

Comment:
```suggestion
        line = next(self.content)

        if line.rstrip() != 'ply':
```

It's better not to mix a conditional with a call that has side effects (such as `next()`) because it might cause bugs when code is refactored at a later time.